### PR TITLE
implemented degradation rate adjustment in sampling probability calculation and updated EM algorithm 

### PR DESCRIPTION
**Pull request by cying111**
- enabled byte compiler in description 
- updated convergence criteria to relative convergence for theta only
- modified sampling matrix with degradation rate
- changed degradation rate calculation to be based on equiRC read counts 
- updated test datasets and codes to make sure test pass
- updated readFromGTF in readWrite script to output exon_rank to avoid the warning message "unknown exon_rank" in tests 
- update style and indentation to multiples of 4 spaces and make sure 80 characters per line including spaces

### CODE REVIEW

File: data-raw/DATASET.R [hunk 1]
```diff
@@ -77,20 +96,233 @@ seCombinedGeneExpected <- transcriptToGeneExpression(seCombined)
...
     seWithDistExpected,
     seGeneExpected, seExtendedGeneExpected,
     seCombinedGeneExpected, seCombinedExtendedGeneExpected,
-    seUCSCExpected,
     internal = TRUE, overwrite = TRUE
 )
+
+
+## inst data creation
+.libPaths("/usr/local/lib/R/site-library")
```
Comment (jonathangoeke) [inline data-raw/DATASET.R:111]: can we remove references to paths?

File: data-raw/DATASET.R [hunk 2]
```diff
@@ -77,20 +96,233 @@ seCombinedGeneExpected <- transcriptToGeneExpression(seCombined)
...
+require(readxl)
+
+
+require(ggplot2)
+require(RColorBrewer)
+require(gridExtra)
+
+
+cat('Setting working directory')
+wkdir <- '/mnt/ont/s3.ontdata.store.genome.sg/projects/'
```
Comment (jonathangoeke) [inline data-raw/DATASET.R:127]: can we change this to a local path?

File: data-raw/DATASET.R [hunk 3]
```diff
@@ -77,20 +96,233 @@ seCombinedGeneExpected <- transcriptToGeneExpression(seCombined)
...
+
+## get gene List
+se <- readRDS("/mnt/ont/s3.ontdata.store.genome.sg/projects/bamboo_output/seOutput2020-04-30_updated_wBC.rds")
```
Comment (jonathangoeke) [inline data-raw/DATASET.R:130]: can we change this to a local path?

File: data-raw/DATASET.R [hunk 4]
```diff
@@ -77,20 +96,233 @@ seCombinedGeneExpected <- transcriptToGeneExpression(seCombined)
...
+txvec <- geneTx[GENEID %in% genevec]$TXNAME
+
+geneList.file <- paste0(wkdir,"/geneList.txt")
+write.table(genevec, file = geneList.file, sep = '\t',
+            row.names = FALSE, col.names = FALSE)
+
+
+
+## get gtf
+gtf.file <- "/mnt/ont/annotations/Grch38/ensembl-91/Homo_sapiens.GRCh38.91.gtf"
```
Comment (jonathangoeke) [inline data-raw/DATASET.R:157]: local path? or directly the URL?

**Review by jonathangoeke — APPROVED** [review]
Thanks for the code edits @cying111 ! I probably need a proper introduction to bioconductor code style 😅 I included some comments on how the data is generated, is it possible to include local paths rather than absolute paths? Can be done later as well.

File: data-raw/DATASET.R (hunk 1 above)
Comment (cying111) [reply data-raw/DATASET.R:111]: yes, we can remove this! this is more for my local usage.

**Pull request merged**